src/lib.rs

```diff
@@ -1,3 +1,22 @@
+#![warn(missing_docs)]
+//! # Motivation and Goals
+//!
+//! 1. Ergonomics is first priority. Being able to prototype quickly and code
```

Comment:
Oh, yeah, I forgot to say that. I need help to write good module-level documentation :D